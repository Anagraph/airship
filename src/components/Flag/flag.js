import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Icon from '../Icon/icon';
import { theme, shadows } from '../../constants';

const Wrapper = styled.div`
  background: ${props => props.theme.ui01};
  display: flex;
  padding: 16px;
  border-radius: 4px;
  box-shadow: ${shadows.shadow16};
  max-width: 260px;

  button {
    border: 0;
    background: transparent;
    box-shadow: none;
    outline: none;
    padding: 0;
    margin: 0;
    cursor: pointer;
    display: flex;
  }
`;
Wrapper.displayName = 'Flag';
Wrapper.defaultProps = {
  theme,
};

const FlagIcon = styled.div`
  flex: 0 0 auto;
`;
Wrapper.displayName = 'Flag.Handle';

const FlagContent = styled.div`
  flex: 1;
  padding: 0 8px;
`;
FlagContent.displayName = 'Flag.Content';

const Flag = ({ children, onClick }) => (
  <Wrapper>
    {children}
    <FlagIcon key="flagButton">
      <button onClick={event => onClick(event)}>
        <Icon icon="close" size={12} />
      </button>
    </FlagIcon>
  </Wrapper>
);

Flag.Icon = FlagIcon;
Flag.Content = FlagContent;

Flag.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func.isRequired,
};

export default Flag;
