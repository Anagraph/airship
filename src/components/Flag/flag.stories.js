import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { ThemeProvider } from 'styled-components';
import Flag from './flag';
import Icon from '../Icon/icon';
import Text from '../Typography/text';
import { theme } from '../../constants';

const CUSTOM_THEME = {
  ...theme,
  ui01: '#333',
  type01: 'white',
};

storiesOf('Flag', module)
  .add('Default', () => (
    <Flag onClick={action('Closed!')}>
      <Flag.Icon>
        <Icon icon="alert_fill" size={12} color="#80B622" />
      </Flag.Icon>
      <Flag.Content>
        <Text weight="medium">
          You are now conected
        </Text>
        <Text>
          You have been added to the group “New Store on this region”
        </Text>
      </Flag.Content>
    </Flag>
  ))
  .add('With custom theme', () => (
    <ThemeProvider theme={CUSTOM_THEME}>
      <Flag onClick={action('Closed!')}>
        <Flag.Icon>
          <Icon icon="alert_fill" size={12} color="#80B622" />
        </Flag.Icon>
        <Flag.Content>
          <Text weight="medium">
            You are now conected
          </Text>
          <Text>
            You have been added to the group “New Store on this region”
          </Text>
        </Flag.Content>
      </Flag>
    </ThemeProvider>
  ));
