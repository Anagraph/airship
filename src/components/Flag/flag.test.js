import React from 'react';
import Flag from './flag';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

describe('render', () => {
  it('renders without crashing', () => {
    const spy = jest.fn();
    const component = renderer.create(
      <Flag onClick={spy}>
        <Flag.Icon>Icon</Flag.Icon>
        <Flag.Content>Content</Flag.Content>
      </Flag>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call callback when changed', () => {
    const spy = jest.fn();
    const component = shallow(
      <Flag onClick={spy}>
        <Flag.Icon>Icon</Flag.Icon>
        <Flag.Content>Content</Flag.Content>
      </Flag>
    );

    const input = component.find('button').first();
    input.simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
