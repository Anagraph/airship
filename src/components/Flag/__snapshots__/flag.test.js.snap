// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders without crashing 1`] = `
.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 12px;
  vertical-align: middle;
  width: 12px;
  fill: #1785FB;
}

.c0 {
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 16px;
  border-radius: 4px;
  box-shadow: 0 4px 16px 0 rgba(44,44,44,0.16);
  max-width: 260px;
}

.c0 button {
  border: 0;
  background: transparent;
  box-shadow: none;
  outline: none;
  padding: 0;
  margin: 0;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0 8px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Icon
  </div>
  <div
    className="c2"
  >
    Content
  </div>
  <div
    className="c1"
  >
    <button
      onClick={[Function]}
    >
      <svg
        className="c3"
        size={12}
        viewBox="0 0 16 16"
      >
        <path
          d="M8,8.58232323 L1.70292632,14.8793969 C1.5421222,15.040201 1.28140721,15.040201 1.12060309,14.8793969 C0.95979897,14.7185928 0.95979897,14.4578778 1.12060309,14.2970737 L7.41767677,8 L1.12060309,1.70292632 C0.95979897,1.5421222 0.95979897,1.28140721 1.12060309,1.12060309 C1.28140721,0.95979897 1.5421222,0.95979897 1.70292632,1.12060309 L8,7.41767677 L14.2970737,1.12060309 C14.4578778,0.95979897 14.7185928,0.95979897 14.8793969,1.12060309 C15.040201,1.28140721 15.040201,1.5421222 14.8793969,1.70292632 L8.58232323,8 L14.8793969,14.2970737 C15.040201,14.4578778 15.040201,14.7185928 14.8793969,14.8793969 C14.7185928,15.040201 14.4578778,15.040201 14.2970737,14.8793969 L8,8.58232323 Z"
        />
      </svg>
    </button>
  </div>
</div>
`;
